### ISSUE TITLE
Return a proper MotionObservable from the operators

### ISSUE THREAD

**Issue by appsforartists** [material-motion/material-motion-experiments-js#209]
Right now, the operators that need to create a new observable use `new this.constructor()` to do so.  This avoids a circular dependency problem (`MotionObservable` depending on `withMotionOperators`, which depends on operators that depend on `MotionObservable`).  Unfortunately, it also means we can't easily add operators to classes whose constructors don't match the `Observable` interface: `MotionProperty` and `MotionSubject` are blocked on this.

The best idea I've come up with so far is to add a `__observableConstructor` property to every class that we want to be able to hang motion operators off of, and then use `new this.__observableConstructor()` from the operators that need it.  This also has the nice side effect of allowing authors to define their own operators and inject them into the system.

**Comment by appsforartists** [comment]
Hey look - I made an issue for this already!

Duplicate of #194
